rewriter/Command.cc

```diff
@@ -85,7 +85,7 @@ void Command::run(core::MutableContext ctx, ast::ClassDef *klass) {
     flags.isSelfMethod = true;
     flags.discardDef = true;
     auto selfCall = ast::MK::SyntheticMethod(call->loc, call->loc, call->name, std::move(newArgs),
-                                             ast::MK::UntypedNil(call->loc), flags);
+                                             ast::MK::RaiseTypedUnimplemented(call->declLoc), flags);
```

Comment:
I was thinking that this wouldn't work because this would lead inference to think the return type of the `call` method is `T.noreturn` and then that would conflict with whatever the user declared as the return type themselves.

However, it does work. Is that because `T.noreturn` is bottom, and thus a subtype of anything the user declared as the return type?

Reply:
> Is that because T.noreturn is bottom, and thus a subtype of anything the user declared as the return type?

yep

Reply:
> Is that because T.noreturn is bottom, and thus a subtype of anything the user declared as the return type?

yep